hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/util/Util.java

```diff
@@ -182,6 +182,32 @@ public static void checkSerializable(Object object, String objectName) {
         }
     }
 
+    /**
+     * Checks that the {@code object} is not null and  implements
+     * {@link Serializable} and is correctly serializable by actually
+     * trying to serialize it. This will reveal some non-serializable
+     * field early.
+     * <p>
+     * Usage:
+     * <pre>{@code
+     * void setValue(@Nonnull Object value) {
+     *     this.value = checkNonNullAndSerializable(value, "value");
+     * }
+     * }</pre>
+     *
+     * @param object     object to check
+     * @param objectName object description for the exception
+     * @return given object
+     * @throws IllegalArgumentException if {@code object} is not serializable
+     */
+    public static <T> T checkNonNullAndSerializable(T object, String objectName) {
```

Comment:
`requireNonNullAndSerializable` would better match the standard `Objects.requireNonNull`. We can also use it inside the method...